sa/sa.go

```diff
@@ -2115,8 +2113,39 @@ func (ssa *SQLStorageAuthority) GetAuthorizations2(ctx context.Context, req *sap
 		return nil, err
 	}
 
+	authz2IDMap := map[int64]bool{}
+	// Once the old authorization storage format fallback is removed we don't need
+	// this length check as if there are none returned we can just return immediately.
+	if len(authzModels) > 0 {
+		// Instead of using a JOIN on the orderToAuthz2 table in order to filter out
+		// authorizations that don't have associated orders we do a second query
+		// and do the filtering in code. This allows us to bypass the large number
+		// of duplicate rows that are generated by using a JOIN.
```

Comment:
I'd like to see a little more description here of the conditions under which we see a large number of rows in the `orderToAuthz2` table for a single authorization, since it's slightly harder to grok, and is the motivating reason for this change. I'd also like to see a similar description in the PR description.